new-dti-website-redesign/src/components/SectionSep.tsx

```diff
@@ -1,22 +1,47 @@
+import XIcon from '../app/design-system/components/icon/XIcon';
+import IconButton from './IconButton';
+
 type SectionSepProps = {
   grid?: boolean;
+  hasX?: boolean;
+  isMobile?: boolean;
+  onClickX?: () => void;
 };
 
-export default function SectionSep({ grid = false }: SectionSepProps) {
+export default function SectionSep({
+  grid = false,
+  hasX = false,
+  isMobile = false,
+  onClickX = () => {}
+}: SectionSepProps) {
+  const boxCount = isMobile ? 8 : 16;
+  const lastIndex = boxCount - 1;
+
   if (!grid) {
     return <div className="w-full h-16 md:h-32 sectionSep border-border-1 border-l-1 border-r-1" />;
   }
 
   return (
     <div className="w-full overflow-hidden sectionSep border-border-1 border-l-1 border-r-1">
       <div className="flex justify-center">
-        {Array.from({ length: 16 }).map((_, i) => (
+        {Array.from({ length: boxCount }).map((_, i) => (
           <div
             key={i}
-            className="aspect-square outline-[0.5px] outline-solid outline-border-1 shrink-0
+            className={`aspect-square outline-[0.5px] outline-solid outline-border-1 shrink-0
                   w-[calc(100%/8)] 
-                  md:w-[calc(100%/16)]"
-          />
+                  md:w-[calc(100%/16)] ${hasX && i === lastIndex ? 'relative' : ''}`}
+          >
+            {hasX && i === lastIndex && (
+              <IconButton
+                className="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 border-none hover:bg-transparent"
```

Comment:
```suggestion
                className="absolute w-full h-full rounded-none border-none"
```

That way we can have the correct hover states

<img width="517" alt="Screenshot 2025-05-31 at 8 20 51 PM" src="https://github.com/user-attachments/assets/7a4dd577-9783-4cb0-9240-0a9cd29826c0" />